### PR TITLE
PYTHON-2388 Begin PyMongo 4.0 migration guide

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Most of the verbiage I copied from migrate-to-pymongo3.rst. Going forward we should update this guide whenever we make a breaking change in 4.0.

### CODE REVIEW

**Review by prashantmital — APPROVED** [review]
Looks good for the most part! We can add a section about `FutureWarning` once we add them to UUID handling.

File: doc/changelog.rst [hunk 1]
```diff
@@ -4,8 +4,15 @@ Changelog
 Changes in Version 4.0
 ----------------------
 
+PyMongo 4.0 brings a number of improvements as well as some backward breaking
+changes. For example, all APIs deprecated in PyMongo 3.X have been removed.
+Be sure to read the changes listed below and the :doc:`migrate-to-pymongo4`
+before upgrading from PyMongo 3.x.
+
+.. warning:: PyMongo 4.0 drops support for Python 2.7, 3.4, and 3.5.
```
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply doc/changelog.rst]: Done.

File: doc/migrate-to-pymongo4.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+The first step in any successful migration involves upgrading to, or
+requiring, at least that latest version of PyMongo 3.x. If your project has a
+requirements.txt file, add the line "pymongo >= 3.12, < 4.0" until you have
+completely migrated to PyMongo 4. Most of the key new methods and options from
+PyMongo 4.0 are backported in PyMongo 3.12 making migration much easier.
+
+.. note:: Users of PyMongo 2.X who wish to upgrade to 4.x must first upgrade
+   to PyMongo 3.x by following the :doc:`migrate-to-pymongo3`.
+
```
[the remark this answers is not in the document]
Comment (ShaneHarvey) [reply doc/migrate-to-pymongo4.rst:28]: Done.

**Review by ShaneHarvey — COMMENTED** [review]
> We can add a section about FutureWarning once we add them to UUID handling.

I agree but I think we should do that in the UUID ticket itself.

**Pull request merged**